opensearchpy/_async/transport.py

```diff
@@ -397,7 +399,8 @@ async def perform_request(
                     ignore=ignore,
                     timeout=timeout,
                 )
-
+                url = quote(url)
```

Comment:
This is done in a loop `for attempt ...`, might fail the second time quoting an already quoted URL.